app/src/pages/Desktop/Protocols/ProtocolPreview/DeckView.tsx

```diff
@@ -107,14 +137,49 @@ export function DeckView(props: DeckViewProps): JSX.Element {
                   addressableArea.id,
                   deckDef.cutoutFixtures
                 )
+                const labwareOnSlot = Object.entries(labware).find(
+                  ([_, lw]) =>
+                    getSlotInLocationStack(lw.stack) === addressableArea.id
+                )
+                const isTiprack =
+                  labwareOnSlot != null
+                    ? labwareEntities[labwareOnSlot[0]].def.parameters.isTiprack
+                    : false
+                const { isActiveLayerVisible } =
+                  labwareOnSlot != null
+                    ? getActiveLayer(
+                        isTiprack,
+                        Object.values(pipettes),
+                        labwareOnSlot[0],
+                        selectedRunTimeCommand
+                      )
+                    : { isActiveLayerVisible: false }
+                let fixtureBaseColor = lightFill
+                if (showDeckRenders) {
+                  if (isActiveLayerVisible) {
+                    fixtureBaseColor = '#E6D5EC'
+                  } else if (
+                    !isActiveLayerVisible &&
+                    selectedSlot === addressableArea.id
+                  ) {
+                    fixtureBaseColor = COLORS.grey40
+                  }
+                }
                 return cutoutId != null ? (
                   <SingleSlotFixture
                     key={addressableArea.id}
                     cutoutId={cutoutId}
                     deckDefinition={deckDef}
                     showExpansion={cutoutId === 'cutoutA1'}
-                    fixtureBaseColor={lightFill}
+                    fixtureBaseColor={fixtureBaseColor}
                     slotClipColor={darkFill}
+                    stroke={
+                      showDeckRenders &&
+                      (hoveredSlot === addressableArea.id ||
+                        selectedSlot === addressableArea.id)
+                        ? COLORS.purple50
+                        : undefined
```

Comment:
```suggestion
                        : none
```

or use `&&`.
for the css module migration, probably it would be good to avoid using `undefined`